trillian_log_api.proto

```diff
@@ -99,6 +99,25 @@ message QueueLeavesResponse {
     repeated QueuedLogLeaf queued_leaves = 2;
 }
 
+message AddSequencedLeafRequest {
+    int64 log_id = 1;
+    LogLeaf leaf = 2;
+}
+
+message AddSequencedLeafResponse {
+    google.rpc.Status status = 2;
+}
+
+message AddSequencedLeavesRequest {
+    int64 log_id = 1;
+    repeated LogLeaf leaves = 2;
+}
+
+message AddSequencedLeavesResponse {
```

Comment:
This comment is open for team discussion as this recommendation caught me by surprise initially. 

Consider **not** implementing the batch version of `AddSequencedLeaf`. 
Rational: 
- Batch methods make per-RPC metrics a deceptive metric for how much load the server is receiving.
- Error recovery for unary RPCs is much easier. 
   - gRPC provides built-in error passing
   - Clients can catch errors early
- Unary RPCs are easier to map onto HTTP / REST methods

You're more familiar with the interior design. Would there be a strong performance benefit to this batch method?

Reply:
If anything batches are more applicable to pre-ordered logs as mirrors won't be very useful until they catch up to the base log. We probably don't want to queue a billion entries one at a time. In any case we don't have to map the whole API to REST. There is a single leaf option that clients are welcome to use.

Reply:
I agree that removing batch API would be a nice simplification. Consider, however, the mirroring case.

There are 2 "phases" of operation:
1. Initialization - load many entries (say, 100M) from the target log.
2. Normal operation - keep up with the target updates.

The rate of updates in the normal operation is not too high, 10-100 qps. Probably the underlying storage could cope with such load even with single-entry updates.
@gdbelvin What is the throughput you need for Key Transparency?

The initialization phase, however, would take a long time in the current design because each query would go straight to storage as a Tree Transaction. One way to mitigate that would be to make single-entry queries asynchronous, and introduce some batching on Trillian or storage side. But I think that adding one more API call is simpler.

Reply:
As discussed with @AlCutter we will keep the batch API because it allows for more flexibility in storage implementation. Removing it would force every impl do their own "mutation pool", as performance is important here.